Lib/unittest/mock.py

```diff
@@ -1760,6 +1761,7 @@ def method(self, *args, **kw):
     '__hash__': lambda self: object.__hash__(self),
     '__str__': lambda self: object.__str__(self),
     '__sizeof__': lambda self: object.__sizeof__(self),
+    '__fspath__': lambda self: "%s/%s" % (type(self).__name__, id(self))
```

Comment:
This will elide all naming information. See:
```
>>> os.fspath(m.absolute)
'MagicMock/140083495922656'
>>> str(m.absolute)
"<MagicMock name='mock.absolute' id='140083495922656'>"
```

Or when a mock is given a name. The id is unique per attribute and probably quite unuseful for the user when debugging issues.


Have you considered:
`lambda self: f"{self._extract_mock_name()}/{id(self)}"`

It would look nicer in some scenarios like:

```
>>> os.fspath(mock.MagicMock(**{"__fspath__.return_value": "abc"}).absolute)
'mock.absolute/140476946856592'
```

But maybe weird for some others like:
```
>>> os.fspath(mock.MagicMock() / 'other' / 'path')
'mock.__truediv__().__truediv__()/140476946879824'
```

Reply:
Agree that naming information should be preserved, what would you think of:
```
lambda self: f"{type(self).__name__}/{self._extract_mock_name()}/{id(self)}"
```

The `__truediv__` case is a bit strange but it is correct, though.